laravel-app/app/Services/PostService.php

```diff
@@ -42,4 +42,26 @@ public function updatePost(UserPost $post, array $validatedData): bool
             return false;
         }
     }
+
+    public function sharePost(UserPost $post): UserPost
+    {
+        $sharedPost = new UserPost();
+        $sharedPost->content = $post->content;
+
+        if ($post->isShared()) {
+            $sharedPost->original_post_id = $post->original_post_id;
+        } else {
+            $sharedPost->original_post_id = $post->id;
+        }
```

Comment:
Prevent using of else in the if statement.

https://thomas-sanctorum.medium.com/how-to-simply-avoid-else-in-php-and-others-programming-language-ccd6a07c4e1

Reply:
done sir, prevented else statement. Thank you ❤️